### PR TITLE
fix(vm): ignore ssd disk flag with virtio interface

### PR DESCRIPTION
**Pull request by abdo-farag**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -2,7 +2,8 @@ GOFMT_FILES?=$$(find . -name '*.go' | grep -v vendor)
 NAME=terraform-provider-proxmox
 TARGETS=darwin linux windows
 TERRAFORM_PLUGIN_EXTENSION=
-VERSION=0.12.0 # x-release-please-version
```
Comment (bpg-dev) [inline Makefile]: could you please revert this change?

File: proxmoxtf/resource_virtual_environment_vm.go [hunk 2]
```diff
@@ -2488,7 +2488,9 @@ func resourceVirtualEnvironmentVMGetDiskDeviceObjects(
 		diskDevice.Size = &sizeString
 		diskDevice.SizeInt = &size
 		diskDevice.IOThread = &ioThread
-		diskDevice.SSD = &ssd
+        if strings.Contains(diskInterface, "scsi") {
```
Comment (bpg-dev) [inline proxmoxtf/resource_virtual_environment_vm.go]: From the PVE docs:
> Note that SSD emulation is not supported on VirtIO Block drives.

so I think this should be reflected in the condition then, i.e. `if !strings.HasPrefix(diskInterface, 'virtio') {...}`.

Also linter doesn't like cobbling `if` statements with unrelated assignments, so you may want to move this `if` block down a little bit, after the line 2494

**Review by bpg — CHANGES_REQUESTED** [review]
Hey @abdo-farag! Thanks for the PR!

I have couple of comments, could you take a look pls?

File: Makefile (hunk 1 above)
Comment (abdo-farag) [reply Makefile]: Hi @bpg 

This space makes some problems with me when building on Linux with zsh.
So when I run init with this space I get error.
``` terraform init``` 
Error: Failed to query available provider packages

Could not retrieve the list of available versions for provider bpg/proxmox: provider registry.terraform.io/bpg/proxmox was not found in any of the search locations

It's wierd somehow. but anyway if it bother you I will remove it.

Comment (bpg-dev) [reply Makefile]: Hmmm, that's interesting. My guess is when you run `make` it sets `VERSION` exactly to `0.12.0 # x-release-please-version` 🤦🏻
Good find, I'll take a look.

**Review by bpg — APPROVED** [review]

**Pull request merged**